File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples/VB/WPFTemplateApp/WPFTemplateApp/DataClasses/Employee.vb
Public Class Employee
  Inherits ObjectBase
#Region "Private Variables"
  Private _EmployeeNumber As Integer = 0
  Private _FirstName As String = String.Empty
  Private _LastName As String = String.Empty
  Private _SSN As String = String.Empty
#End Region

#Region "Public Properties"
  Public Property EmployeeNumber() As Integer
    Get
      Return _EmployeeNumber
    End Get
    Set(ByVal value As Integer)
      If _EmployeeNumber <> value Then
        _EmployeeNumber = value
        RaisePropertyChanged("EmployeeNumber")
      End If
    End Set
  End Property

  Public Property FirstName() As String
    Get
      Return _FirstName
    End Get
    Set(ByVal value As String)
      If _FirstName <> value Then
        _FirstName = value
        RaisePropertyChanged("FirstName")
      End If
    End Set
  End Property

  Public Property LastName() As String
    Get
      Return _LastName
    End Get
    Set(ByVal value As String)
      If _LastName <> value Then
        _LastName = value
        RaisePropertyChanged("LastName")
      End If
    End Set
  End Property

  Public Property SSN() As String
    Get
      Return _SSN
    End Get
    Set(ByVal value As String)
      If _SSN <> value Then
        _SSN = value
        RaisePropertyChanged("SSN")
      End If
    End Set
  End Property
#End Region
End Class

File: _wpf/wpf-business-programmer/3-wpf-business-programmer-m3-exercise-files/Samples/CSharp/WPFTemplateApp/WPFTemplateApp/DataClasses/Employee.cs

namespace WPFTemplateApp
{
  public class Employee : ObjectBase
  {
    #region Private Variables
    private int _EmployeeNumber = 0;
    private string _FirstName = string.Empty;
    private string _LastName = string.Empty;
    private string _SSN = string.Empty;
    #endregion

    #region Public Properties
    public int EmployeeNumber
    {
      get { return _EmployeeNumber; }
      set
      {
        if (_EmployeeNumber != value)
        {
          _EmployeeNumber = value;
          RaisePropertyChanged("EmployeeNumber");
        }
      }
    }

    public string FirstName
    {
      get { return _FirstName; }
      set
      {
        if (_FirstName != value)
        {
          _FirstName = value;
          RaisePropertyChanged("FirstName");
        }
      }
    }
    
    public string LastName
    {
      get { return _LastName; }
      set
      {
        if (_LastName != value)
        {
          _LastName = value;
          RaisePropertyChanged("LastName");
        }
      }
    }

    public string SSN
    {
      get { return _SSN; }
      set
      {
        if (_SSN != value)
        {
          _SSN = value;
          RaisePropertyChanged("SSN");
        }
      }
    }
    #endregion
  }
}
